specifications/api/swagger.yaml

```diff
@@ -1862,6 +1959,138 @@ components:
+        - balance
+        - assets
+        - delegation
+        - state
+        - tip
+      properties:
+        id: *walletId
+        name: *walletName
+        account_index: *derivationSegment
+        address_pool_gap: *walletAddressPoolGap
+        passphrase: *walletPassphraseInfo
+        payment_script_template: *scriptTemplate
+        delegation_script_template: *scriptTemplate
+        balance: *walletBalance
+        assets: *walletAssetsBalance
+        delegation: *ApiWalletDelegation
+        state: *walletState
+        tip: *blockReference
+
+    ApiPendingSharedWallet: &ApiPendingSharedWallet
+      type: object
+      required:
+        - id
+        - name
+        - account_index
+        - address_pool_gap
+        - payment_script_template
+      properties:
+        id: *walletId
+        name: *walletName
+        account_index: *derivationSegment
+        address_pool_gap: *walletAddressPoolGap
+        payment_script_template: *scriptTemplate
+        delegation_script_template: *scriptTemplate
+
+    ApiSharedWallet: &ApiSharedWallet
+      type: object
+      oneOf:
+        - title: Pending shared wallet (collecting account public keys for each co-signer)
+          <<: *ApiPendingSharedWallet
+        - title: Active shared wallet (account public keys for each co-signers is collected)
+          <<: *ApiActiveSharedWallet
+
+    ApiSharedWalletPostDataFromMnemonics: &ApiSharedWalletPostDataFromMnemonics
+      type: object
+      required:
+        - name
+        - mnemonic_sentence
+        - passphrase
+        - account_index
+        - payment_script_template
+      properties:
+        name: *walletName
+        mnemonic_sentence: *walletMnemonicSentence
+        mnemonic_second_factor: *walletSecondFactor
+        passphrase: *walletPassphrase
+        account_index: *derivationSegment
+        payment_script_template: *scriptTemplate
+        delegation_script_template: *scriptTemplate
+
+    ApiSharedWalletPostDataFromAccountPubX: &ApiSharedWalletPostDataFromAccountPubX
+      type: object
+      required:
+        - name
+        - account_public_key
+        - account_index
+        - payment_script_template
+      properties:
+        name: *walletName
+        account_public_key: *walletAccountPubkey
+        account_index: *derivationSegment
+        payment_script_template: *scriptTemplate
+        delegation_script_template: *scriptTemplate
+
+    ApiSharedWalletPostData: &ApiSharedWalletPostData
+      type: object
+      oneOf:
+        - title: Create shared wallet from mnemonics
+          <<: *ApiSharedWalletPostDataFromMnemonics
+        - title: Create shared wallet from account public key
+          <<: *ApiSharedWalletPostDataFromAccountPubX
+
+    ApiSharedWalletPatchData: &ApiSharedWalletPatchData
+      type: object
+      required:
+        - cosigner
+        - account_public_key
+      properties:
+        cosigner: *cosigner
+        account_public_key: *walletAccountPubkey
```

Comment:
Looking at the integration test scenario, this stoke me as a little bit inconsistent with how the accounts are defined. What you think about defining the patch as:

```yaml
ApiSharedWalletPatchData: *cosigners
```

So that, the update of cosigners is done similarly to how they are initially defined. That is, one would patch something like:

```json
{ "cosigner#0": "1423856bc91c49e928f6f30f4e8d665d53eb4ab6028bd0ac971809d514c92db11423856bc91c49e928f6f30f4e8d665d53eb4ab6028bd0ac971809d514c92db1" }
```

To update the `cosigner#0`. It carries the same information, but I find it more consistent / transparent that way.

Reply:
very sensible -> done